packages/x-components/src/x-modules/next-queries/store/actions/fetch-and-save-next-query-preview.action.ts

```diff
@@ -0,0 +1,43 @@
+import { NextQueriesXStoreModule } from '../types';
+import { cancellablePromise, CancelSymbol } from '../../../../utils/cancellable-promise';
+
+let cancelPromise: undefined | (() => void);
+
+const fetchAndSave: NextQueriesXStoreModule['actions']['fetchAndSaveNextQueryPreview'] = (
+  { dispatch, commit },
+  nextQuery
```

Comment:
I'm fine with this being either a `query: string` or a `nextQuery: NextQuery`. But I don't think it is a good idea to mix both types and say `nextQuery: string`, as it is confusing for the reader. Indeed in the type definition you named it `query`.